common/src/hash.rs

```diff
@@ -0,0 +1,104 @@
+use crate::*;
+
+impl ToHash256 for Member {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for BlockHeader {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for Transaction {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for Agenda {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for AgendaProof {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for ExtraAgendaTransaction {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for ChatLog {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for GenesisInfo {
+    fn to_hash256(&self) -> Hash256 {
+        Hash256::hash(serde_json::to_vec(self).unwrap())
+    }
+}
+
+impl ToHash256 for Commit {
+    fn to_hash256(&self) -> Hash256 {
+        match self {
+            Commit::Block(x) => x.to_hash256(),
+            Commit::Transaction(x) => x.to_hash256(),
+            Commit::Agenda(x) => x.to_hash256(),
+            Commit::AgendaProof(x) => x.to_hash256(),
+            Commit::ExtraAgendaTransaction(x) => x.to_hash256(),
+            Commit::ChatLog(x) => x.to_hash256(),
+        }
+    }
+}
+
+impl Transaction {
+    /// Returns the alternative hash of the transaction, which is for the Merkle tree.
```

Comment:
Instead of alternative hash, how about body hash or something else?

Reply:
the fact that it hashes the body is not necessarily for being exposed in the interface.

Reply:
in the interface, that it's for the Merkle tree is what the user should care.

Reply:
Understood.